sherpa/astro/ui/utils.py

```diff
@@ -3788,8 +3790,10 @@ def _save_type(self, objtype, id, filename, bkg_id=None, **kwargs):
         if bkg_id is not None:
             d = self.get_bkg(id, bkg_id)
 
-        if type(d) in (sherpa.astro.data.DataIMG, sherpa.astro.data.DataIMGInt,
-                       sherpa.data.Data2D, sherpa.data.Data2DInt):
+        if isinstance(d, (sherpa.astro.data.DataIMG,
```

Comment:
Would it be easier (and more future proof) to simply test the number of axes, i.e. the spacial dimension of the independent variable?

Reply:
This is a completely-undocumented function that I don't understand the pre- or post-conditions for use, or even what it is meant to be doing, so I am loath to change it significantly at this time!